### PR TITLE
Feature/basic fracturing

### PR DESCRIPTION
**Pull request by Developer-The-Great**

### CODE REVIEW

File: legion/engine/core/math/geometry.hpp [hunk 1]
```diff
@@ -173,9 +173,18 @@ namespace legion::core::math
...
+                std::cout << "\t\t\t\t\t\t\t" << " point q is on triangle surface " << math::to_string(q) << std::endl;
+            }
+           
             return projectionLength;
         }
 
+        if (debug)
+        {
+            std::cout << "\t\t\t\t\t\t\t" << " point q is NOT on triangle surface " << math::to_string(q) << std::endl;
+        }
```
Comment (Algo-ryth-mix) [inline legion/engine/core/math/geometry.hpp]: these couts should go

**Review by Algo-ryth-mix — CHANGES_REQUESTED** [review]
you simulatenously need more and less comments. Please remove commented code and add a few more doxygen comments!

File: legion/engine/core/math/geometry.hpp (hunk 1 above)
Comment (Ragingram2) [reply legion/engine/core/math/geometry.hpp]: yeah these should be log::debugs.

File: legion/engine/physics/colliders/convexcollider.cpp [hunk 2]
```diff
@@ -43,17 +47,30 @@ namespace legion::physics
...
+       
+        PhysicsStatics::DetectConvexConvexCollision(this,convexCollider, 
+            manifold.transformA, manifold.transformB, convexCollisionInfo, manifold);*/
+
+        /*if (!manifold.isColliding)
+        {
+            return;
+        }*/
+
+
```
Comment (Ragingram2) [inline legion/engine/physics/colliders/convexcollider.cpp:70]: is this still used?

File: legion/engine/physics/colliders/convexcollider.cpp [hunk 3]
```diff
@@ -107,7 +127,30 @@ namespace legion::physics
...
+        //    std::make_shared< ConvexConvexPenetrationQuery>(convexCollisionInfo.ARefFace.ptr
+        //        , convexCollisionInfo.BRefFace.ptr, worldFaceCentroidA, worldFaceNormalA, convexCollisionInfo.ARefSeperation, true);
+
+        //auto baPenetrationQuery =
+        //    std::make_shared < ConvexConvexPenetrationQuery>(convexCollisionInfo.BRefFace.ptr, convexCollisionInfo.ARefFace.ptr,
+        //        worldFaceCentroidB, worldFaceNormalB, convexCollisionInfo.BRefSeperation, false);
+
+        //auto abEdgePenetrationQuery =
+        //    std::make_shared < EdgePenetrationQuery>(convexCollisionInfo.edgeRef.ptr, convexCollisionInfo.edgeInc.ptr, worldEdgeAPosition, worldEdgeNormal,
+        //        convexCollisionInfo.aToBEdgeSeperation, false);
```
Comment (Ragingram2) [inline legion/engine/physics/colliders/convexcollider.cpp:153]: will any of this be used?

File: legion/engine/physics/colliders/convexcollider.cpp [hunk 4]
```diff
@@ -148,8 +189,617 @@ namespace legion::physics
...
+        minMaxWorldAABB = PhysicsStatics::ConstructAABBFromTransformedVertices
+        (vertices, transform);
+
+        //math::vec3 min = std::get<0>(minMaxWorldAABB);
+        //math::vec3 max = std::get<1>(minMaxWorldAABB);
+        //math::vec3 difference = max - min;
+
+      /*  debug::user_projectDrawLine(min, min + math::vec3(difference.x,0,0), math::colors::red, 5.0f);
+        debug::user_projectDrawLine(min, min + math::vec3(0,difference.y,0), math::colors::red, 5.0f);
+        debug::user_projectDrawLine(min, min + math::vec3(0,0,difference.z), math::colors::red, 5.0f);*/
```
Comment (Ragingram2) [inline legion/engine/physics/colliders/convexcollider.cpp]: same here

File: legion/engine/physics/data/physics_manifold_precursor.h [hunk 5]
```diff
@@ -13,6 +13,8 @@ namespace legion::physics
 		ecs::component_handle<physicsComponent> physicsComponentHandle;
 		int id;
 
+
```
Comment (Ragingram2) [inline legion/engine/physics/data/physics_manifold_precursor.h:16]: shouldn't this be an hpp

File: legion/engine/physics/physics_contact.hpp [hunk 6]
```diff
@@ -44,6 +44,8 @@ namespace legion::physics
 
         int contactCount = 0;
 
+        
+
```
Comment (Ragingram2) [inline legion/engine/physics/physics_contact.hpp]: thank you github, thank you very much

**Review by Ragingram2 — APPROVED** [review]
There is some commented code that I'm not sure if it will be used or not. I know that is it won't be used it will mostly likely be cleaned up in the second run of this. So I am approving

File: legion/engine/physics/colliders/convexcollider.cpp (hunk 3 above)
Comment (Developer-The-Great) [reply legion/engine/physics/colliders/convexcollider.cpp:153]: yes i tried to refactor this part and something when wrong along the way, gonna keep it here for now

File: legion/engine/physics/colliders/convexcollider.cpp (hunk 2 above)
Comment (Developer-The-Great) [reply legion/engine/physics/colliders/convexcollider.cpp:70]: also a failed attempt at a refactor

**Review by Algo-ryth-mix — APPROVED** [review]

**Pull request merged**